pkg/labels/labels.go

```diff
@@ -81,6 +81,25 @@ func (ls *Labels) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// MatchLabels returns a subset of Labels that matches/does not match with the provided label names based on the 'on' boolean.
+// If on is set to true, it returns the subset of labels that match with the provided label names and its inverse when 'on' is set to false.
+func (ls Labels) MatchLabels(on bool, names ...string) Labels {
```

Comment:
I think this method may be unnecessary for this PR.

Reply:
Thoughts ?

Reply:
Yeah, how about just using `Labels.Has()` instead of the custom set and then putting the rest directly into the branch where you handle the many-to-many error?

Reply:
> then putting the rest directly into the branch where you handle the many-to-many error?

I don't think I understand this

Reply:
Also using `Labels.Has()` we assume that `matching.On` will be true. We also have to account for the other case correct?

Reply:
Good point. Yeah, you probably still want the `nameSet` then.

Regarding the confusion: I thought your initial question was about whether you really need this method (since it's only needed in one place and somewhat special) or whether you could just put all of that code in the place where you use it (in `engine.go` when that error occurs). So that's what I meant.

Reply:
Got it! Thanks for clarifying. 

I guess keeping this method is okay then? I guess MatchLabels is faster but at the cost of space for maintaining a temporary set.

Reply:
I don't have strong feelings either way regarding keeping the method or putting all that code into the place where you use it, so do whatever you like most :) You already have unit tests for the method now, so...

We really don't care about anything regarding speed or efficiency here since it's not a hot path, but just used for that one error message.

Reply:
Sure! I can vote for keeping the method unless another prometheus maintainer objects :)